app/components/post/index.js

```diff
@@ -6,36 +6,39 @@ import {bindActionCreators} from 'redux';
 
 import {createPost, deletePost, removePost} from 'mattermost-redux/actions/posts';
 import {getCurrentChannelId, isCurrentChannelReadOnly} from 'mattermost-redux/selectors/entities/channels';
-import {getPost} from 'mattermost-redux/selectors/entities/posts';
+import {getPost, makeGetCommentCountForPost} from 'mattermost-redux/selectors/entities/posts';
 import {getCurrentUserId, getCurrentUserRoles} from 'mattermost-redux/selectors/entities/users';
 import {getMyPreferences, getTheme} from 'mattermost-redux/selectors/entities/preferences';
 import {getCurrentTeamUrl, getCurrentTeamId} from 'mattermost-redux/selectors/entities/teams';
-import {canDeletePost, canEditPost, isPostFlagged} from 'mattermost-redux/utils/post_utils';
+import {canDeletePost, canEditPost, isPostFlagged, isSystemMessage} from 'mattermost-redux/utils/post_utils';
 import {isAdmin as checkIsAdmin, isSystemAdmin as checkIsSystemAdmin} from 'mattermost-redux/utils/user_utils';
 
 import {insertToDraft, setPostTooltipVisible} from 'app/actions/views/channel';
 import {addReaction} from 'app/actions/views/emoji';
 import {getDimensions} from 'app/selectors/device';
+import {Posts} from 'mattermost-redux/constants';
 
 import Post from './post';
 
 function mapStateToProps(state, ownProps) {
     const post = getPost(state, ownProps.postId);
-
     const {config, license} = state.entities.general;
     const roles = getCurrentUserId(state) ? getCurrentUserRoles(state) : '';
     const myPreferences = getMyPreferences(state);
     const currentUserId = getCurrentUserId(state);
     const currentTeamId = getCurrentTeamId(state);
     const currentChannelId = getCurrentChannelId(state);
+    const getCommentCountForPost = makeGetCommentCountForPost();
```

Comment:
I believe you need to change the `mapStateToProps` to a parent `makeMapStateToProps` and have this in the parent function like [this one](https://github.com/mattermost/mattermost-mobile/blob/d0d91038576f8469dc7c88035e47f3583131d68f/app/components/post_list/index.js#L15)